vm/entrypoint/src/syscalls/host.rs

```diff
@@ -3,25 +3,36 @@ use core::arch::asm;
 
 /// Call a function in a foreign program.
 ///
-/// `address` is the callee address, `input` is a bytearray to be passed to the
-/// callee function, and `len` is the number of bytes to read from the input bytearray.
+/// `address` is the callee address, `input_ptr` is a bytearray to be passed to the
+/// callee function, and `input_len` is the number of bytes to read from the input bytearray.
+/// `method_ptr` is the name of the function to call, and `method_len` is the length of
+/// the method name in bytes.
 /// `amount` is the number of coins to transfer to the callee.
 /// For now there is no return value and no return status code. The caller can assume
 /// that, if this function returns, the call was successful.
 ///
 /// See https://github.com/athenavm/athena/issues/5 for more information.
 #[allow(unused_variables)]
 #[no_mangle]
-pub extern "C" fn call(address: *const u32, input: *const u32, len: usize, amount: *const u32) {
+pub extern "C" fn call(
```

Comment:
I don't understand why all syscall helpers are marked as `extern "C"`, do you know? They are not called over FFI. They are compiled to risc-v and called normally.

Reply:
We inherited this from sp1, see, e.g.: https://github.com/succinctlabs/sp1/blob/dev/crates/zkvm/entrypoint/src/syscalls/memory.rs

My best guess why this is the case is that some of these syscalls plug directly into the Rust ABI: https://github.com/rust-lang/rust/blob/56e35a5dbb37898433a43133dff0398f46d577b8/library/std/src/sys/pal/zkvm/abi.rs

Reply:
It seems that in our case, the syscalls are exported in: https://github.com/athenavm/athena/blob/ae54d14c7c8b489240765cafb393d5266665cdc6/vm/lib/src/lib.rs#L6-L13 to avoid cyclic dependencies between vm/entrypoint and vm/lib crates... The entrypoint depends on lib so lib cannot depend on entrypoint directly, where these syscalls are defined hence it declares that they "exist somewhere" using `export "C"`. IMHO we should really untangle these kind of things as it makes the code complicated and hard to reason about :(

Reply:
#117